asllib/Parser.mly

```diff
@@ -443,18 +449,20 @@ let untyped_decl_item :=
 (* ------------------------------------------------------------------------- *)
 (* Statement helpers *)
 
-let local_decl_keyword :=
+let local_decl_keyword_non_var :=
   | LET       ; { LDK_Let       }
   | CONSTANT  ; { LDK_Constant  }
-  (* We can't have VAR here otherwise there is a conflict (grammar is not LR1).
+  (* Var is inlined inside stmt as it has differing production choices
   | VAR       ; { LDK_Var       }
   *)
 
-let storage_keyword ==
+let storage_keyword_non_var :=
```

Comment:
I've done this, I've also taken the liberty of doing a bit of naming alignment and ran a few `sed` scripts:
 * renamed `\Nstoragekeyword` to `\Nglobaldeclkeyword`
 * renamed `\buildstoragekeyword` to `\buildglobaldeclkeyword`
 * updated the link definitions and uses for the above to use globaldeclkeyword instead of storagekeyword
 * updated ASTRule.StorageKeyword to ASTRule.GlobalDeclKeyword everywhere.
 
 I've also added `expr_pattern`'s empty record case (my bad I didn't realize patterns have their own file, I was wondering where expr_pattern was, I assumed there was prose signifying that `expr_pattern` builds the same as `expr` in all cases)